src/pages/header.tsx

```diff
@@ -0,0 +1,22 @@
+import { Button, Text, VStack } from "@chakra-ui/react";
+import { useAuth } from "../hooks/useAuth";
+import { useRouter } from "next/router";
+import { Layout } from "../components/Layout";
+
+const HeaderPage = () => {
+  const router = useRouter();
+  const userAuth = useAuth();
```

Comment:
I think you can just destructure it

```
const { logout } = useAuth();
```

Or rename it. Auth is almost always related to user, so 
```
const auth = useAuth()
```

would be much better